Update to golem js 3.0 and TE 2.0

SewerynKras opened the pull request.

SewerynKras committed:
feat: update react package to use golem-js 3.0

SewerynKras committed:
docs: update example vite app to use the new api

SewerynKras committed:
feat: update react package to use task executor 2.0 BETA

SewerynKras committed:
docs: update README

SewerynKras committed:
chore:  remove debug logs

mgordel reviewed (approved):
Locally tested. Everything seems ok

SewerynKras committed:
feat: update task executor to version 2.0

BREAKING CHANGE: the `useExecutor` hook now uses the same options as the new 2.0 task executor
